### PR TITLE
Dockerize App

### LINKED ISSUE #5: Set up relational database
Set up a PostgreSQL database on the VM. This does not include data, only figuring out how to set up the database server.

### PR DESCRIPTION
**Pull request by Duckling92**
Adds Docker images for frontend, backend and database

Adds docker-compose configuration for starting app

Adds CI build test and a simple end-to-end test

Additionally, I propose to move all test-jobs into the same workflow (it appears this is how Github intends it to be...) such that the simple end-to-end test can depend on the unittest jobs. I don't think this is possible with separate workflows.

Closes #5 and #6.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    image: postgres:16-alpine
+    restart: always
+    container_name: terastore-postgres
+    ports:
+      - 5432:5432
+    volumes:
+      - db:/var/lib/postgresql/data
+
+volumes:
+  db:
```
Comment (ehrhorn) [inline docker-compose.yml:40]: Are these empty statements intentional?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      - name: Test app
+        run: npm --prefix frontend/ run test
+  docker-test-build:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    needs: [backend-test, frontend-test]
+    env:
+      POSTGRES_USER: pg_user
+      POSTGRES_PASSWORD: pg_user_pw
+      POSTGRES_DB: terastore-db
```
Comment (ehrhorn) [inline .github/workflows/tests.yml:135]: Is it neccessary to use secrets instead?

File: docker-compose.yml (hunk 1 above)
Comment (Duckling92) [reply docker-compose.yml:40]: Yes, I think it is required to persistently store the volume

File: .github/workflows/tests.yml (hunk 2 above)
Comment (Duckling92) [reply .github/workflows/tests.yml:135]: Don't know. My idea was that it is merely for spinning up a testing environment, so it doesn't matter. For now, I'd say leave it like this.

**Review by ehrhorn — APPROVED** [review]
All good.

**Pull request merged**